lib/common/xml.c

```diff
@@ -2139,14 +2142,13 @@ pcmk__xe_delete_match(xmlNode *xml, xmlNode *search)
 static void
 replace_node(xmlNode *old, xmlNode *new)
 {
-    new = xmlCopyNode(new, 1);
-    pcmk__mem_assert(new);
+    // Pass old for its doc; it won't remain the parent of new
+    new = pcmk__xml_copy(old, new);
+    old = xmlReplaceNode(old, new);
```

Comment:
Are you sure old doesn't remain the parent of new?

Also the returned old could be NULL on memory error

Reply:
As sure as I am of anything when coding. `xmlReplaceNode()` unlinks `new` (separating it from `old`'s subtree); inserts it where `old` was with all the same linkages; unlinks `old`; and returns `old`.

How could the returned `old` be `NULL`?

Reply:
> As sure as I am of anything when coding. `xmlReplaceNode()` unlinks `new` (separating it from `old`'s subtree); inserts it where `old` was with all the same linkages; unlinks `old`; and returns `old`.

Ah, I overlooked that new would be unlinked too
 
> How could the returned `old` be `NULL`?

"Returns: @old or NULL if arguments are invalid or a memory allocation failed"

Reply:
> "Returns: @old or NULL if arguments are invalid or a memory allocation failed"

Ah, I didn't look at the doc comment. I was thinking the body couldn't do that, but yeah `xmlSetTreeDoc()` can fail to strdup...

Reply:
We could either return early or `CRM_ASSERT()` that it succeeded. I'm thinking we should assert, since we often do that for libxml2 memory allocation and since error handling would be tricky here.